backend/src/controllers/reports/genericReportController.js

```diff
@@ -0,0 +1,120 @@
+const {
+  getDocumentApiBody,
+  getReportAndSetRequestHeaders,
+  pdfConfig,
+  validateQueryParameters,
+} = require("./helpers");
+
+// Constants
+const what = { single: "report", plural: "reports" };
+
+/**
+ *
+ * USING THIS CONTROLLER:
+ *
+ * How to change the route: (pseudocode)
+ * EXAMPLE ROUTE: backend/src/routes/reports/Tab_50_rpt_PF_NetRecoverySummaryByQuarter.js
+ *
+ * const getControllerFrom = require("@controllers/reports/genericReportController");
+ * const name = `{whatever your report name is}`;
+ * const controller = getControllerFrom(name);
+ *
+ * How to change the model: (export the data instead of the query - pseudocode)
+ * EXAMPLE MODEL: backend/src/models/reports/Tab_50_rpt_PF_NetRecoverySummaryByQuarter.js
+ *
+ * async({some named parameter(s)}) => {
+ * const [{all teh data for the report }] = await Promise.all([
+ * {list of the knex query promises}
+ * ]);
+ *
+ * return {
+ * <all the data for your report, organized into whatever structure you need>
+ * < you can use `groupByProperty()` from from helpers to add structure for reports 'by fiscal' or 'by portfolio'>
+ * }}
+ *
+ */
+
+/**
+ * Retrieves a controller from the specified filename.
+ *
+ * @param   {string} filename - The name of the file.
+ * @returns {object}          - The controller object.
+ */
+const getControllerFrom = (filename) => {
+  const model = require(`@models/reports/${filename}`);
+  const controller = useController(model, what);
+
+  const getReport = getReportAndSetRequestHeaders();
+
+  const reportHandler = async (request, reply) => {
+    controller.userRequires(request, "PMO-Reports-Capability", reply);
+
+    try {
+      const { templateType, fiscal, portfolio } = validateQueryParameters(request.query);
+      const modifiedGetReport = getReportAndSetRequestHeaders(templateType);
+      controller.getReport = modifiedGetReport;
+
+      const result = await getDataFromModel({ fiscal, portfolio, model });
+      await sendToCdogs({ result, filename, templateType, request });
+
+      if (result) {
+        return result;
+      } else {
+        return controller.noQuery(reply, `There was a problem looking up this Report.`);
+      }
+    } catch (err) {
+      return controller.failedQuery(reply, err, what);
+    }
+  };
+
+  return {
+    ...controller,
+    getReport,
+    [filename]: reportHandler,
+  };
+};
+
+/**
+ * Retrieves data from a model based on fiscal or portfolio information.
+ *
+ * @param   {object}               params           - The parameters for retrieving data.
+ * @param   {string|null}          params.fiscal    - The fiscal information.
+ * @param   {string|Array[string]} params.portfolio - The portfolio information.
+ * @param   {object}               params.model     - The model to retrieve data from.
+ * @returns {Promise}                               - A promise that resolves with the retrieved data.
+ */
+const getDataFromModel = async ({ fiscal = null, portfolio = null, model }) => {
```

Comment:
What if we pass in the parameters into this function.. 
For example 
```suggestion
const getDataFromModel = async (...reportParams) => {
```

and then the model would have a generic function something like getModelData(...reportParams)

Reply:
good idea, I'll simply pass the query and use Destructure to grab the fiscal etc. from inside the model. that way the generic controller stays agnostic about what args are sent to the model.